packages/block-library/src/latest-posts/edit.native.js

```diff
@@ -171,27 +164,24 @@ class LatestPostsEdit extends Component {
 						label={ __( 'Display post date' ) }
 						checked={ displayPostDate }
 						onChange={ this.onSetDisplayPostDate }
-						separatorType="none"
 					/>
 				</PanelBody>
 
-				<PanelBody title={ __( 'Sorting and filtering' ) }>
-					<QueryControls
-						{ ...{ order, orderBy } }
-						numberOfItems={ postsToShow }
-						categoriesList={ categoriesList }
-						selectedCategoryId={
-							undefined !== categories ? Number( categories ) : ''
-						}
-						onOrderChange={ this.onSetOrder }
-						onOrderByChange={ this.onSetOrderBy }
-						onCategoryChange={
-							// eslint-disable-next-line no-undef
-							__DEV__ ? this.onSetCategories : undefined
-						}
-						onNumberOfItemsChange={ this.onSetPostsToShow }
-					/>
-				</PanelBody>
+				<QueryControls
```

Comment:
I don't like that I need to move `PanelBody` to the `QueryControls` for mobile which is different for the web but the mechanism that i implemented will not work in that configuration. I iterate through `PanelBody` children and set the 
 correct separator to them. In that case, the only child is `QueryControls` which breaks the logic :( This is the only one restriction in that implementation, maybe someone has better idea?

Reply:
I see what you mean. I suppose we cannot arbitrarily flatten children of `PanelBody`, so we cannot use components which are a collection of Controls (like `QueryControls`) to compose as children of `PanelBody` (the "grandchildren" won't get separators).

> Note: I could change the QuertyControls to the function that returns an array of controls instead but it will be also different than the web but feels more reusable since now the PanelBody has set the title in QueryControls.

:thinking: I wonder, do you mean _invoking_ `QueryControls` rather than using it as a component? E.g. something like:
```jsx
<PanelBody>{QueryControls({numberOfItems: postsToShow, etc.})}</PanelBody>
```

instead of:

```jsx
<PanelBody><QueryControls numberOfItems={ postsToShow } etc. /></PanelBody>
```

Reply:
> I wonder, do you mean invoking QueryControls rather than using it as a jsx element? E.g. something like:

Yes, exactly, but finally i decided to move the `PanelBody` to the `Controls`.  I'm open to suggestions here :)